framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -669,9 +685,10 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 		// Delete blockHeaders
 		const blockHeaders = await this._chainConnectorStore.getBlockHeaders();
 
-		await this._chainConnectorStore.setBlockHeaders(
-			blockHeaders.filter(blockHeader => blockHeader.height >= this._lastCertificate.height),
+		const updatedBlockHeaders = blockHeaders.filter(
+			blockHeader => blockHeader.height >= this._lastCertificate.height,
```

Comment:
why can't we filter with strict inequality `blockHeader.height > this._lastCertificate.height`?

Reply:
we need blockHeader at last certificate height for certificate generation